### ISSUE TITLE
NullPointerException by CryptoConditionReader.readFulfillment()

### ISSUE THREAD

**Issue by ManujSubhankar** [interledger/java-crypto-conditions#69]
> CryptoConditionReader.readFulfillment()

throws NullPointerException.

Below is the sample code where in I am encoding a Fulfillment and reading it back using CryptoConditionReader.readFulfillment(), which throws NullPointerException.

```
import java.security.InvalidKeyException;
import java.security.KeyPair;
import java.security.MessageDigest;
import java.security.NoSuchAlgorithmException;
import java.security.Signature;
import java.security.SignatureException;

import org.interledger.cryptoconditions.Fulfillment;
import org.interledger.cryptoconditions.der.CryptoConditionReader;
import org.interledger.cryptoconditions.der.DEREncodingException;
import org.interledger.cryptoconditions.types.Ed25519Sha256Fulfillment;

import net.i2p.crypto.eddsa.EdDSAEngine;
import net.i2p.crypto.eddsa.EdDSAPublicKey;
import net.i2p.crypto.eddsa.KeyPairGenerator;

public class Bug {
	
	public static void main(String[] args) throws SignatureException, NoSuchAlgorithmException, InvalidKeyException, DEREncodingException {
		KeyPairGenerator kpg= new KeyPairGenerator();
		KeyPair kp= kpg.generateKeyPair();
		
		MessageDigest md= MessageDigest.getInstance("SHA-512");
		Signature sign= new EdDSAEngine(md);
		sign.initSign(kp.getPrivate());
		sign.update("Hello_world".getBytes());
		
		Ed25519Sha256Fulfillment ffill= new Ed25519Sha256Fulfillment((EdDSAPublicKey) kp.getPublic(), sign.sign());
		
		Fulfillment fill2= CryptoConditionReader.readFulfillment(ffill.getEncoded());
		
	}
}
```

**Comment by sappenin** [comment]
Hi @ManujSubhankar, thanks for reporting this bug.

I have a pretty substantial PR (#68) that tries to fixup the project quite a bit. If you have time, I'd be curious to hear if this issue still exists in the `features/immutable-crypto-conditions` branch.

Otherwise, I'll try to take a look once I have some spare cycles. Thanks!

**Comment by sappenin** [comment]
Just a follow-up here: the `features/immutable-crypto-conditions` branch that I alluded to in my previous comment has now been merged into master, and is now part of version `0.3.0-SNAPSHOT`.

**Comment by sappenin** [comment]
This is verified to be fixed as of version 0.3.0-SNAPSHOT of this library. This issue will be closed when #81 (which adds more formalized tests for this issue) is merged.
